### PR TITLE
reduce the main trace width to 72 columns

### PR DESCRIPTION
**Pull request by grjte**
This PR removes the row address column from the hash chiplet, since we can use the `clk` column from the system component for the same purpose. This reduces the width of the chiplets to 17 columns, which restores the overall width of the main execution trace back to 72 columns

- [x] update docs (constraints, images, etc)
- [x] update processor
- [x] update AIR
- [x] update tests
- [x] disable recursive verifier test
- [x] update zipped excalidraw image files in the mdbook assets

Note:
- Since we're planning to update the trace width quite a bit, we may want to consider disabling the recursive verifier test and updating the verifier constants in a few weeks after all of the changes are made. Alternatively, is there a way we can set the trace columns in a single place so that one correction will resolve all updates (maybe this already exists and I just didn't see it in a cursory look).
- I think we may be able to simplify the hash memoization a bit with these changes, but we have an open issue to revisit this anyway, and I didn't want to complicate this PR

### CODE REVIEW

**Comment by grjte** [comment]
Here's what running fibonacci for 2^20 cycles looks like before and after. The full proof time dropped by 2% from 9485 ms to 9282 ms.

## BEFORE
Generated execution trace of 73 columns and 1048576 steps in 397 ms
Built domain of 2^23 elements in 8 ms
Extended execution trace of 73 columns from 2^20 to 2^23 steps (8x blowup) in 2302 ms
Computed execution trace commitment (Merkle tree of depth 23) in 1063 ms
Built auxiliary trace segment of 9 columns and 2^20 steps in 43 ms
Extended execution trace of 9 columns from 2^20 to 2^23 steps (8x blowup) in 686 ms
Computed execution trace commitment (Merkle tree of depth 23) in 393 ms
Evaluated constraints over domain of 2^23 elements in 2983 ms
Converted constraint evaluations into 8 composition polynomial columns of degree 1048575 in 191 ms
Evaluated 8 composition polynomial columns over LDE domain (2^23 elements) in 412 ms
Computed constraint evaluation commitment (Merkle tree of depth 23) in 314 ms
Built DEEP composition polynomial of degree 1048574 in 358 ms
Evaluated DEEP composition polynomial over LDE domain (2^23 elements) in 66 ms
Computed 4 FRI layers from composition polynomial evaluations in 79 ms
Determined 27 query positions in 1 ms
Built proof object in 76 ms
Program with hash 055af6c034549f6e3cb885a789906c3c970c2a64c45a99f2fabf87b1fe665751 proved in 9485 ms

## AFTER
Generated execution trace of 72 columns and 1048576 steps in 381 ms
Built domain of 2^23 elements in 9 ms
Extended execution trace of 72 columns from 2^20 to 2^23 steps (8x blowup) in 2153 ms
Computed execution trace commitment (Merkle tree of depth 23) in 983 ms
Built auxiliary trace segment of 9 columns and 2^20 steps in 43 ms
Extended execution trace of 9 columns from 2^20 to 2^23 steps (8x blowup) in 704 ms
Computed execution trace commitment (Merkle tree of depth 23) in 392 ms
Evaluated constraints over domain of 2^23 elements in 3005 ms
Converted constraint evaluations into 8 composition polynomial columns of degree 1048575 in 198 ms
Evaluated 8 composition polynomial columns over LDE domain (2^23 elements) in 413 ms
Computed constraint evaluation commitment (Merkle tree of depth 23) in 291 ms
Built DEEP composition polynomial of degree 1048574 in 383 ms
Evaluated DEEP composition polynomial over LDE domain (2^23 elements) in 66 ms
Computed 4 FRI layers from composition polynomial evaluations in 80 ms
Determined 27 query positions in 0 ms
Built proof object in 48 ms
Program with hash 055af6c034549f6e3cb885a789906c3c970c2a64c45a99f2fabf87b1fe665751 proved in 9282 ms

**Review by bobbinth — APPROVED** [review]
Looks good! Thank you! I left one minor nit inline. Let's merge this after @Al-Kindi-0 re-enables the STARK verifier test.

File: stdlib/asm/crypto/stark/random_coin.masm [hunk 1]
```diff
@@ -75,119 +73,103 @@ end
...
 
-    # Compute parameters and save them to memory
-    # ---------------------------------------------------------------------------------------------
-
-    # Save `log(trace_length)`, `num_queries`, and `grinding_factor` to memory
+    # Save the parameters in memory for later use
     dup exec.constants::trace_length_log_ptr mem_store
     dup.1 exec.constants::number_queries_ptr mem_store
     dup.3 exec.constants::grinding_factor_ptr mem_store
-    #=> [log(trace_length), num_queries, log(blowup), grinding, ...] (9 cycles)
```
Comment (hackaugusto) [inline stdlib/asm/crypto/stark/random_coin.masm:90]: This reverted documentation changes introduced by https://github.com/0xPolygonMiden/miden-vm/pull/973 , was that intentional?

I was relying on that to understand the stack and implement https://github.com/0xPolygonMiden/miden-vm/pull/960

Comment (Al-Kindi-0) [reply stdlib/asm/crypto/stark/random_coin.masm:90]: I probably removed it by mistake as I don't see now a reason for removing it

Comment (bobbinth) [reply stdlib/asm/crypto/stark/random_coin.masm:90]: It was probably me who removed it by mistake because I screwed up commit history at some point.

**Pull request merged**